internal/pkg/agent/vault/vault_linux.go

```diff
@@ -157,3 +224,40 @@ func deriveKey(pw []byte, salt []byte) ([]byte, []byte, error) {
 func (v *Vault) filepathFromKey(key string) string {
 	return filepath.Join(v.path, fileNameFromKey(v.key, key))
 }
+
+// writeFile "atomic" file write, utilizes temp file and replace
+// os.CreateTemp creates the file with 0600 mask, which is what we need
+func writeFile(fp string, data []byte) (err error) {
+	dir, fn := filepath.Split(fp)
+	if dir == "" {
+		dir = "."
+	}
+
+	f, err := os.CreateTemp(dir, fn)
+	if err != nil {
+		return fmt.Errorf("failed creating temp file: %w", err)
+	}
+	defer func() {
```

Comment:
[Suggestion]
1 - move the comment about the `os.CreateTem` to where it's used.
2 - I think the sentence could be improved

```suggestion
// writeFile performs an "atomic" file write operation, using a temporary file
// and then replacing the original.
func writeFile(fp string, data []byte) (err error) {
	dir, fn := filepath.Split(fp)
	if dir == "" {
		dir = "."
	}

	// os.CreateTemp creates the file with a 0600 permission mask,
	// which is the required permission setting.
	f, err := os.CreateTemp(dir, fn)
	if err != nil {
		return fmt.Errorf("failed creating temp file: %w", err)
	}
	defer func() {
```